layouts/_default/baseof.html

```diff
@@ -0,0 +1,113 @@
+<!DOCTYPE html>
+<html lang="{{ .Site.Language.Lang }}" dir="{{ .Site.Language.LanguageDirection | default "ltr" }}">
+<head>
+    {{ partial "docs/html-head" . }}
+    {{ partial "docs/inject/head" . }}
+
+    {{/* This injects a class based on either cssClass in the _index.md or the filepath. 
+        (The cssClass override is especially useful when the filepath name isn't a valid CSS class name) */}}
+
+    {{ $injectedClass :=  "" }}
+    {{ $currentLink := .Permalink }}
+    {{ $bookSection := default "docs" .Site.Params.BookSection  }}
+
+    {{ if eq $bookSection "*" }}
+        {{ $bookSection = "/" }}
+    {{ end }}
+
+    {{ with .Site.GetPage $bookSection }}
+        {{ range (where .Pages "Params.bookhidden" "ne" true) }}
+            {{ $page := . }}
+            {{ $filePath := replace .File.Path "\\" "/"}}
+            {{ $parts := split $filePath "/" }}
+            {{ $section := index $parts 1 }}
+            {{ $section = replace $section ".md" "" }}
+            {{ if in $currentLink $section }}
+                {{ with $page.Params.cssClass }}
+                    {{ $injectedClass = $page.Params.cssClass }}
+                {{ else }}
+                    {{ $injectedClass = $section }}
+                {{ end }}
+            {{ end }}
+        {{ end }}
+    {{ end }}
```

Comment:
I think we can probably use the hugo trick for [returning a value from a partial](https://gohugo.io/templates/partials/#returning-a-value-from-a-partial) to move this code out of this file/make it more reusable.

Refactoring this into `/layouts/partials/platen/getCssClass.html` should allow us to rewrite this section of the file as:

```
{{ $injectedClass := partial "platen/getCssClass.html" . }}
```

Reply:
Between this and reducing the injection points, that would leave our patch to this file as +1 line added, 1 line changed.

Reply:
Relatedly, I think we _might_ want this file in two commits:

1. Add the `baseof.html` exactly as it exists in hugo-book right now
2. Add our changes on top.

This will make future diffing cleaner/easier and may enable us to do absolutely silly pants strategems for keeping this file automatically up to date that I'm mulling over.

Reply:
Makes total sense. I'll avoid the partial implementation for this PR then and stick to the cssClass/injectedClass stuff in this one.

Reply:
Confirmed that with the following in `/layouts/partials/hugo/getCssClass.html`:

```html
{{/*
  This injects a class based on either cssClass in the _index.md or the filepath.
  (The cssClass override is especially useful when the filepath name isn't a valid CSS class name)
*/}}
{{- $injectedClass :=  "" -}}
{{- $currentLink := .Permalink -}}
{{- $bookSection := default "docs" .Site.Params.BookSection -}}
{{- if eq $bookSection "*" -}}
  {{- $bookSection = "/" -}}
{{- end -}}
{{- with .Site.GetPage $bookSection -}}
  {{- range (where .Pages "Params.bookhidden" "ne" true) -}}
    {{- $page := . -}}
    {{- $filePath := replace .File.Path "\\" "/" -}}
    {{- $parts := split $filePath "/" -}}
    {{- $section := index $parts 1 -}}
    {{- $section = replace $section ".md" "" -}}
    {{- if in $currentLink $section -}}
      {{- with $page.Params.cssClass -}}
        {{- $injectedClass = $page.Params.cssClass -}}
      {{- else -}}
        {{- $injectedClass = $section -}}
      {{- end -}}
    {{- end -}}
  {{- end -}}
{{- end -}}
{{- return $injectedClass -}}
```

Reply:
> Makes total sense. I'll avoid the partial implementation for this PR then and stick to the cssClass/injectedClass stuff in this one.

I think we'll still need it in this PR, we'll just need it in a third discrete commit in the PR. If you like, we can pair on this? I'm around for a while yet @binarystargames

Reply:
Sure, that works too. Do you intend to bring baseof.html in in a separate PR then?

Reply:
https://binarystar.games/docs/3rd-party/nova-burnout/ 

Confirming that page-level cssClass is working on my end. I'll push the change I made (and the body-only changes) to the repo.

Reply:
Yeah, I'm a bit stumped on the partial bit. Keep getting a "partial that returns a value needs a non-zero argument" error. I tried setting a default value for $injectedClass but that didn't help.